pkg/setting/setting_unified_alerting.go

```diff
@@ -109,6 +111,13 @@ type UnifiedAlertingSettings struct {
 	NotificationLogRetention time.Duration
 }
 
+type RecordingRuleSettings struct {
+	URL               string
+	BasicAuthUsername string
+	BasicAuthPassword string
+	TenantID          string
```

Comment:
This doesn't match the nesting in the `.ini` config. Purposefully kept this simple but left the config a bit more structured since that'll be harder to change later. We could flatten that, or more deeply nest this, if we wanted them to match.